server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -322,24 +320,3 @@ func (ds *cachedMysql) DeleteTeam(ctx context.Context, teamID uint) error {
 
 	return nil
 }
-
-func (ds *cachedMysql) ListScheduledQueriesForAgents(ctx context.Context, teamID *uint) ([]*fleet.Query, error) {
-	var teamIDVal uint
-	if teamID != nil {
-		teamIDVal = *teamID
-	}
-
-	key := fmt.Sprintf(scheduledQueriesForAgentsKey, teamIDVal)
-	if x, found := ds.c.Get(key); found {
-		if queries, ok := x.([]*fleet.Query); ok {
```

Comment:
Maybe a silly question, but is not obvious to me why this was the bottle neck? Was it because `[]*fleet.Query ` was always to big to keep in the cache layer? And the I/O overhead was too much?

Reply:
By default (i.e. unless the thing cloned implements the `Cloner` interface) the cloning is done by reflection using a 3rd party package we use and it's quite slow, especially in a hot path. I was honestly a bit surprised that this was (likely) the main cause, but it makes sense given that the traces in the load testing did not show a particular query being slow and the DB was not under stress, so something in the Go code was causing slowness.

If it turns out we need some caching at some point, we can use a custom type and implement a hand-written cloning method as we did on AppConfig, but from the load tests it's not obvious that this would benefit from being cached.

Reply:
e.g. you can see for `ListScheduledQueriesInPack`, a custom type has been implemented for the cached layer so that it has a custom cloning method:

https://github.com/fleetdm/fleet/blob/main/server/fleet/scheduled_queries.go#L59-L86